src/IO/H5/Python/File.cpp

```diff
@@ -0,0 +1,51 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <boost/python.hpp>
+#include <boost/python/reference_existing_object.hpp>
+#include <boost/python/return_value_policy.hpp>
+#include <boost/python/suite/indexing/vector_indexing_suite.hpp>
+#include <hdf5.h>
+#include <sstream>
+#include <string>
+#include <utility>
+
+#include "IO/H5/Dat.hpp"
+#include "IO/H5/File.hpp"
+#include "IO/H5/Header.hpp"
+#include "IO/H5/OpenGroup.hpp"
+#include "IO/H5/Type.hpp"
+#include "IO/H5/Version.hpp"
+#include "PythonBindings/VectorPyList.hpp"
+
+namespace bp = boost::python;
+
+namespace py_bindings {
+void bind_h5file() {
+  // Wrapper for basic H5File operations
+  bp::class_<h5::H5File<h5::AccessType::ReadWrite>, boost::noncopyable>(
+      "H5File", bp::init<std::string, bool>())
+      .def("name",
+           +[](const h5::H5File<h5::AccessType::ReadWrite>& f) {
+             return f.name();
+           })
+      .def("get_dat",
```

Comment:
Can you make the accessors `__getitem__` and `__setitem__` instead (again, to closely mirror h5py)?

Reply:
Sorry for the long response time @nilsleiffischer, I think that implementing `__getitem__` makes sense to me, but I'm a bit confused about implementing `__setitem__` since this isn't exactly analogous to adding a group in h5py, as the `insert_dat` function does more than just adding a group to the file.

Reply:
It seems to me most "Pythonic" implementation would be to implement `__setitem__` such that you can hand it any pre-constructed object, e.g. a `Dat`, and, if needed, do a run-time check for its type (instead of the the template specialisations that we have in the C++).
Since right now I'm not sure we actually need the setter right now, we can also defer this to a future PR.

Reply:
I don't think we should do the "Pythonic" thing. In my experience when people claim something is the "Pythonic" thing to do, it's the wrong thing to do for various different reasons. I'm not really sold by using the "Pythonic" argument. Additionally, I think there is a lot of merit to having the python interface behave like the C++ interface.